pkg/providers/imagefamily/bootstrap/heplers.go

```diff
@@ -0,0 +1,181 @@
+		return "kubenet"
+	default:
+		return ""
+	}
+}
+
+func getStringFromNetworkPolicyType(enum nbcontractv1.NetworkPolicyType) string {
+	switch enum {
+	case nbcontractv1.NetworkPolicyType_NETWORK_POLICY_TYPE_AZURE:
+		return "azure"
+	case nbcontractv1.NetworkPolicyType_NETWORK_POLICY_TYPE_CALICO:
+		return "calico"
+	default:
+		return ""
+	}
+}
+
+func getStringFromLoadBalancerSkuType(enum nbcontractv1.LoadBalancerSku) string {
+	switch enum {
+	case nbcontractv1.LoadBalancerSku_LOAD_BALANCER_SKU_BASIC:
+		return "Basic"
+	case nbcontractv1.LoadBalancerSku_LOAD_BALANCER_SKU_STANDARD:
+		return "Standard"
+	default:
+		return ""
+	}
+}
+
+func getBoolFromFeatureState(state nbcontractv1.FeatureState) bool {
+	return state == nbcontractv1.FeatureState_FEATURE_STATE_ENABLED
+}
+
+func getBoolStringFromFeatureState(state nbcontractv1.FeatureState) string {
+	return strconv.FormatBool(state == nbcontractv1.FeatureState_FEATURE_STATE_ENABLED)
+}
+
+func getBoolStringFromFeatureStatePtr(state *nbcontractv1.FeatureState) string {
+	if state == nil {
+		return "false"
+	}
+
+	if *state == nbcontractv1.FeatureState_FEATURE_STATE_ENABLED {
+		return "true"
+	}
+
+	return "false"
+}
+
+// deref is a helper function to dereference a pointer of any type to its value
+func deref[T interface{}](p *T) T {
+	return *p
+}
+
+func getStringifiedMap(m map[string]string, delimiter string) string {
+	result := strings.Join(lo.MapToSlice(m, func(k, v string) string {
+		return fmt.Sprintf("%s=%s", k, v)
+	}), delimiter)
+	return result
+}
+
+func getStringifiedStringArray(arr []string, delimiter string) string {
+	if len(arr) == 0 {
+		return ""
+	}
+
+	return strings.Join(arr, delimiter)
+}
+
+// getKubenetTemplate returns the base64 encoded Kubenet template.
+func getKubenetTemplate() string {
+	return base64.StdEncoding.EncodeToString(kubenetTemplateContent)
+}
+
+// getSysctlContent returns the base64 encoded sysctl content.
+func getSysctlContent() string {
+	return base64.StdEncoding.EncodeToString(sysctlTemplateContent)
+}
+
+func getContainerdConfig(nbcontract *nbcontractv1.Configuration) string {
+	if nbcontract == nil {
+		return ""
+	}
+
+	containerdConfig, err := containerdConfigFromNodeBootstrapContract(nbcontract)
+	if err != nil {
+		return fmt.Sprintf("error getting containerd config from node bootstrap variables: %v", err)
+	}
+
+	return base64.StdEncoding.EncodeToString([]byte(containerdConfig))
+}
+
+func containerdConfigFromNodeBootstrapContract(nbcontract *nbcontractv1.Configuration) (string, error) {
+	if nbcontract == nil {
+		return "", fmt.Errorf("node bootstrap contract is nil")
+	}
+
+	var buffer bytes.Buffer
+	if err := containerdConfigTemplateForNBContract.Execute(&buffer, nbcontract); err != nil {
+		return "", fmt.Errorf("error executing containerd config template for NBContract: %w", err)
+	}
+
+	return buffer.String(), nil
+}
+
+func getIsMIGNode(gpuInstanceProfile string) bool {
```

Comment:
I created an example of a function to get IsMIGNode status. It simply checks if gpuInstanceProfile is empty or not. With this, we should be able to further remove IsMIGNode from the contract.
Feel free to use this to resolve GPUConfig.IsMIGNode if needed. If you find a better way to do it, feel free to remove this function.